src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/InfrastructureFormula.kt

```diff
@@ -123,6 +143,25 @@ class InfrastructureFormula<out T : VirtualUsers>(
         is MulticastVirtualUsersFormula -> MulticastVirtualUsersFormula.Builder(virtualUsersFormula).network(network).build()
         else -> virtualUsersFormula
     } as VirtualUsersFormula<T>
+
+    class Builder<out T : VirtualUsers>(
+        private val investment: Investment,
+        private val jiraFormula: JiraFormula,
+        private val virtualUsersFormula: VirtualUsersFormula<T>,
+        private val aws: Aws
```

Comment:
The more required parameters, the more it hurts to use it 🤕.
We can add sensible defaults for everything except for `aws` (due to its impure constructor).
E.g. we can use `StandaloneFormula`, `StackVirtualUsersFormula` and a generic `Investment`.
Providing a `StandaloneFormula` hurts a bit due to the 3 params (🤕). It could easily have a zero-arg builder.
Providing a SVUF hurts only a little, due to 1 mandatory param. But there's no sensible default, so we'll have to ask for this param too.

All of the above can be done later, by providing a new `InfrastructureFormula.Builder` constructor, so I'm ok with merging as is, so it's your call.

Reply:
It's problematic with SVUF because it's generic. But I've added the others.